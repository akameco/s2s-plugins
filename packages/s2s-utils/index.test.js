// @flow
import { getImportPath, template } from '.'

test('getImportPath same folder', () => {
  const result = getImportPath('path/to/index.js', 'path/to/test.js')
  expect(result).toBe('./test')
})

test('getImportPath same folder when not ext', () => {
  const result = getImportPath('path/to/index', 'path/to/test')
  expect(result).toBe('./test')
})

test('getImportPath parent folder', () => {
  const result = getImportPath('path/to/nest/index.js', 'path/to/test.js')
  expect(result).toBe('../test')
})

test('template', () => {
  const ast = template(`type Action = A`)()
  expect(ast).toMatchSnapshot()
})
