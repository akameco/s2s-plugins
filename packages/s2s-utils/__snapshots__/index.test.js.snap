// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot defaultImport 1`] = `
Object {
  "source": Object {
    "type": "StringLiteral",
    "value": "source",
  },
  "specifiers": Array [
    Object {
      "local": Object {
        "name": "local",
        "type": "Identifier",
      },
      "type": "ImportDefaultSpecifier",
    },
  ],
  "type": "ImportDeclaration",
}
`;

exports[`snapshot typeImport 1`] = `
Object {
  "importKind": "type",
  "source": Object {
    "type": "StringLiteral",
    "value": "source",
  },
  "specifiers": Array [
    Object {
      "imported": Object {
        "name": "imported",
        "type": "Identifier",
      },
      "local": Object {
        "name": "local",
        "type": "Identifier",
      },
      "type": "ImportSpecifier",
    },
  ],
  "type": "ImportDeclaration",
}
`;

exports[`template 1`] = `
Node {
  "__clone": undefined,
  "_fromTemplate": true,
  "end": undefined,
  "id": Node {
    "__clone": undefined,
    "_fromTemplate": true,
    "end": undefined,
    "loc": undefined,
    "name": "Action",
    "start": undefined,
    "type": "Identifier",
  },
  "loc": undefined,
  "right": Node {
    "__clone": undefined,
    "_fromTemplate": true,
    "end": undefined,
    "id": Node {
      "__clone": undefined,
      "_fromTemplate": true,
      "end": undefined,
      "loc": undefined,
      "name": "A",
      "start": undefined,
      "type": "Identifier",
    },
    "loc": undefined,
    "start": undefined,
    "type": "GenericTypeAnnotation",
    "typeParameters": null,
  },
  "start": undefined,
  "type": "TypeAlias",
  "typeParameters": null,
}
`;
